sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AlterTableExec.scala

```diff
@@ -44,3 +46,35 @@ case class AlterTableExec(
     Seq.empty
   }
 }
+
+/**
+ * Physical plan node for adding a check constraint with validation.
+ */
+case class AddCheckConstraintExec(
+    catalog: TableCatalog,
+    ident: Identifier,
+    change: TableChange,
+    condition: String,
+    child: SparkPlan) extends V2CommandExec with UnaryExecNode {
+
+  override def output: Seq[Attribute] = Seq.empty
+
+  override protected def run(): Seq[InternalRow] = {
+    if (child.executeTake(1).nonEmpty) {
+      throw QueryExecutionErrors.newCheckViolation(
+        condition, ident.toQualifiedNameParts(catalog).quoted)
```

Comment:
`.quote` does quoting if needed, but `toSqlId` always quotes. Let's follow the existing pracise: pass `Seq[String]` to `newCheckViolation` and call `toSqlId` there.

Reply:
`toSqlId`  exists in `DataTypeErrorsBase`. It's weird to used it on query execution error.